### ISSUE TITLE
tempalates for openstack should not CamelCase

### ISSUE THREAD

**Issue by wallnerryan** [sparkleformation/sparkle_formation#214]
```
➜  openstack-sparkle sfn print -f nova_compute
{
  "HeatTemplateVersion": "2015-04-30",
  "Resources": {
    "MyInstance": {
      "Type": "OS::Nova::Server",
      "Properties": {
        "KeyName": "demo",
        "Image": "CentOS-7-1702",
        "Flavor": "m1.medium"
      }
    }
  }
}
➜  openstack-sparkle cat sparkleformation/nova_compute.rb
SparkleFormation.new(:nova_compute, :provider => :open_stack) do

  heat_template_version '2015-04-30'

  resources.my_instance do
    type 'OS::Nova::Server'
    properties.key_name 'demo'
    properties.image 'CentOS-7-1702'
    properties.flavor 'm1.medium'
  end

end

➜  openstack-sparkle sfn create sparkle-nova  -f nova_compute -u
[DEBUG]: Initial compile parameters - {}
[DEBUG]: Merged compile parameters - {}
[Sfn]: SparkleFormation: create
[Sfn]:   -> Name: sparkle-nova
{"explanation": "The server could not comply with the request since it is either malformed or otherwise incorrect.", "code": 400, "error": {"message": "Template not in valid format: Template format version not found.", "traceback": null, "type": "HTTPBadRequest"}, "title": "Bad Request"}
ERROR: Miasma::Error::ApiError::RequestError: Bad Request - 400: Template not in valid format: Template format version not found.
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/miasma-0.3.2/lib/miasma/types/api.rb:127:in `block in request'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/bogo-0.2.8/lib/bogo/retry.rb:65:in `run!'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/miasma-0.3.2/lib/miasma/types/api.rb:149:in `retryable_request'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/miasma-0.3.2/lib/miasma/types/api.rb:123:in `request'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/miasma-open-stack-0.1.4/lib/miasma/contrib/open_stack/orchestration.rb:42:in `stack_save'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/miasma-0.3.2/lib/miasma/models/orchestration/stack.rb:116:in `perform_save'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/miasma-0.3.2/lib/miasma/models/orchestration/stack.rb:109:in `save'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/sfn-3.0.22/lib/sfn/command/create.rb:64:in `block in execute!'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/sfn-3.0.22/lib/sfn/command_module/callbacks.rb:19:in `api_action!'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/sfn-3.0.22/lib/sfn/command/create.rb:63:in `execute!'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/sfn-3.0.22/bin/sfn:52:in `block (4 levels) in <top (required)>'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/bogo-cli-0.2.12/lib/bogo-cli/setup.rb:26:in `block in bogo_cli_run'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/slop-3.6.0/lib/slop.rb:260:in `parse!'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/slop-3.6.0/lib/slop.rb:235:in `parse!'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/slop-3.6.0/lib/slop.rb:65:in `parse!'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/slop-3.6.0/lib/slop.rb:54:in `parse'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/bogo-cli-0.2.12/lib/bogo-cli/setup.rb:48:in `define'
/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/sfn-3.0.22/bin/sfn:14:in `<top (required)>'
/Users/rwallner/.rvm/gems/ruby-2.3.3/bin/sfn:22:in `load'
/Users/rwallner/.rvm/gems/ruby-2.3.3/bin/sfn:22:in `<main>'
/Users/rwallner/.rvm/gems/ruby-2.3.3/bin/ruby_executable_hooks:15:in `eval'
/Users/rwallner/.rvm/gems/ruby-2.3.3/bin/ruby_executable_hooks:15:in `<main>'
```

If i modify the `JSON` to use lower snake case and use `openstack client` it works

```                                                                                                                                                                                              
(oskolla) ➜  phase1-openstack cat heat-t2.json
{
  "heat_template_version": "2015-04-30",
  "resources": {
    "MyInstance": {
      "type": "OS::Nova::Server",
      "properties": {
        "key_name": "demo",
        "image": "CentOS-7-1702",
        "flavor": "m1.medium"
      }
    }
  }
}

(oskolla) ➜  phase1-openstack openstack stack create -f json -t heat-t2.json test2
{
  "description": "No description",
  "stack_status_reason": "Stack CREATE started",
  "stack_name": "test2",
  "creation_time": "2017-04-13T15:49:58Z",
  "updated_time": null,
  "stack_status": "CREATE_IN_PROGRESS",
  "id": "d09996ec-fd59-4de9-a6f7-229b156bc402"
}%           

```

**Comment by wallnerryan** [comment]
modifying `/Users/rwallner/.rvm/gems/ruby-2.3.3/gems/sparkle_formation-3.0.20/lib/sparkle_formation/sparkle_struct.rb` line `51` to `@_camel_keys = false` fixed this for me.

```
➜  openstack-sparkle sfn print -f nova_compute
{
  "heat_template_version": "2015-04-30",
  "resources": {
    "my_instance": {
      "type": "OS::Nova::Server",
      "properties": {
        "key_name": "demo",
        "image": "CentOS-7-1702",
        "flavor": "m1.medium"
      }
    }
  }
}
```

Was able to create the server too. I saw the docs about CamelCase here http://www.sparkleformation.io/docs/sparkle_formation/sparkleformation-dsl.html but not sure what needs to be set to get the above results, setting or using the documented did not work nor did trying to set the `AttributeStruct.camel_keys = false` anywhere in the template or config

**Comment by chrisroberts** [comment]
Was fixed in changes withing #217
